tiledb/sm/array/array.cc

```diff
@@ -381,11 +407,17 @@ Status Array::close() {
   array_schemas_all_.clear();
   metadata_.clear();
   metadata_loaded_ = false;
-  is_open_ = false;
+  set_array_closed();
 
   return Status::Ok();
 }
 
+void Array::set_array_closed() {
```

Comment:
For clarity, IMO we should add a note here (and maybe in the .h as well) that the sentry is also released upon destruction.